### PR TITLE
Style guide

### PR DESCRIPTION
**Pull request by scottgonzalez**
Adds JSCS and JSHint for linting, via `grunt lint`. Uses JSCS for automated whitespace fixes, via `grunt jscs:fix`.

### CODE REVIEW

File: src/PointerEvent.js [hunk 1]
```diff
@@ -58,9 +58,9 @@ function PointerEvent(inType, inDict) {
 
   // define inherited MouseEvent properties
   // skip bubbles and cancelable since they're set above in initEvent()
-  for(var i = 2, p; i < MOUSE_PROPS.length; i++) {
-    p = MOUSE_PROPS[i];
-    e[p] = inDict[p] || MOUSE_DEFAULTS[i];
+  for (var i = 2, p; i < MOUSE_PROPS.length; i++) {
+    p = MOUSE_PROPS[ i ];
```
Comment (csnover) [inline src/PointerEvent.js]: oh, this whitespace change seems wrong.

**Comment by arschmitz** [comment]
I have the updates for the single / double quotes all ready once this lands.

**Comment by scottgonzalez** [comment]
I just pushed an update that disallows spaces inside brackets. @arschmitz @csnover Is this good to land now?

File: .jscsrc [hunk 2]
```diff
@@ -0,0 +1,74 @@
+{
```
Comment (jzaefferer) [inline .jscsrc]: Can't you use one of the existing presets as a starting point? That should massively reduce the amount of config here.

Comment (arschmitz) [inline .jscsrc]: @scottgonzalez correct me if i'm wrong but isn't this just the jQuery preset with the bracket and paren spacing rules changed?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
+You may also want to take a look at our [commit & pull request guide](http://contribute.jquery.org/commits-and-pull-requests/) and [style guides](http://contribute.jquery.org/style-guide/)* for instructions on how to maintain your fork and submit your code. Before we can merge any pull request, we'll also need you to sign our [contributor license agreement](http://contribute.jquery.org/cla/).
 
 You can find us on [IRC](http://irc.jquery.org), specifically in #pep should you have any questions. If you've never contributed to open source before, we've put together [a short guide with tips, tricks, and ideas on getting started](http://contribute.jquery.org/open-source/).
+
+----
+
+\* PEP follows the [jQuery JavaScript Style Guide](http://contribute.jquery.org/style-guide/js/) with the following exception:
+
+* There are no spaces inside parenthesized bodies.
```
Comment (jzaefferer) [inline CONTRIBUTING.md]: I don't know what "parenthesized bodies" means. Its not used in the jQuery style guide, its not an AST term...

File: package.json [hunk 4]
```diff
@@ -21,10 +21,13 @@
...
     "esperanto": "^0.6.6",
-    "intern": "2.2.2",
     "grunt": "~0.4.1",
+    "grunt-contrib-jshint": "0.11.2",
     "grunt-contrib-uglify": "^0.4.0",
-    "grunt-git-authors": "^3.0.0"
+    "grunt-git-authors": "^3.0.0",
+    "grunt-jscs": "1.8.0",
+    "intern": "2.2.2",
+    "jscs": "1.12.0"
```
Comment (scottgonzalez) [inline package.json]: Oops. That's because I started with just jscs. I'll remove it.

File: .jscsrc (hunk 2 above)
Comment (scottgonzalez) [inline .jscsrc]: I tried that, but some of the overrides didn't seem to work, complaining that they could only be turned on, not off. Perhaps I was doing something wrong. I then started by copying the jquery preset, but it also seemed to be missing a ton of rules, so I ended up just going through every rule and picking the ones that applied.

File: CONTRIBUTING.md (hunk 3 above)
Comment (scottgonzalez) [inline CONTRIBUTING.md]: It's anything that appears inside parens. We didn't want to list all of the technical terms.

File: .jscsrc (hunk 2 above)
Comment (jzaefferer) [inline .jscsrc]: You have to set them to `null`. Hopefully the error messages one day will make that more obvious...

Comment (scottgonzalez) [inline .jscsrc]: Ok, so perhaps we should start with a PR to get the jquery preset to be more comprehensive.

Comment (jzaefferer) [inline .jscsrc]: That would be useful for all our projects.

File: CONTRIBUTING.md (hunk 3 above)
Comment (jzaefferer) [inline CONTRIBUTING.md]: "There are no spaces around anything that appears inside parens." works for me.

**Pull request closed without merging**